File: NYTPhotoViewerTests/NYTPhotosOverlayViewTests.m
//
//  NYTPhotosOverlayViewTests.m
//  NYTPhotoViewer
//
//  Created by Brian Capps on 2/26/15.
//  Copyright (c) 2015 NYTimes. All rights reserved.
//

@import UIKit;
@import XCTest;

#import <NYTPhotoViewer/NYTPhotosOverlayView.h>

@interface NYTPhotosOverlayViewTests : XCTestCase
@end

@implementation NYTPhotosOverlayViewTests

- (void)testNavigationBarExistsAfterInitialization {
    NYTPhotosOverlayView *overlayView = [[NYTPhotosOverlayView alloc] init];
    XCTAssertNotNil(overlayView.navigationBar);
}

- (void)testLeftBarButtonItemSetterAffectsNavigationBar {
    NYTPhotosOverlayView *overlayView = [[NYTPhotosOverlayView alloc] init];
    UIBarButtonItem *leftBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone target:nil action:nil];
    overlayView.leftBarButtonItem = leftBarButtonItem;
    XCTAssertEqualObjects(leftBarButtonItem, [overlayView.navigationBar.items.firstObject leftBarButtonItem]);
}

- (void)testRightBarButtonItemSetterAffectsNavigationBar {
    NYTPhotosOverlayView *overlayView = [[NYTPhotosOverlayView alloc] init];
    UIBarButtonItem *rightBarButtonItem = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone target:nil action:nil];
    overlayView.rightBarButtonItem = rightBarButtonItem;
    XCTAssertEqualObjects(rightBarButtonItem, [overlayView.navigationBar.items.firstObject rightBarButtonItem]);
}

- (void)testTitleSetterAffectsNavigationBar {
    NYTPhotosOverlayView *overlayView = [[NYTPhotosOverlayView alloc] init];
    NSString *title = @"title";
    overlayView.title = title;
    XCTAssertEqualObjects(title, [overlayView.navigationBar.items.firstObject title]);
}

- (void)testTitleTextAttributesSetterAffectsNavigationBar {
    NYTPhotosOverlayView *overlayView = [[NYTPhotosOverlayView alloc] init];
    NSDictionary *titleTextAttributes = @{NSForegroundColorAttributeName: [UIColor orangeColor]};
    overlayView.titleTextAttributes = titleTextAttributes;
    XCTAssertEqualObjects(titleTextAttributes, overlayView.navigationBar.titleTextAttributes);
}

@end

File: Example/NYTPhotoViewer-SwiftTests/NYTPhotosOverlayViewTests.swift
//
//  NYTPhotosOverlayViewTests.swift
//  NYTPhotoViewer
//
//  Created by Mark Keefe on 3/21/15.
//  Copyright (c) 2015 The New York Times. All rights reserved.
//

import XCTest
import UIKit

class NYTPhotosOverlayViewTests: XCTestCase {

    func testNavigationBarExistsAfterInitialization() {
        let overlayView = NYTPhotosOverlayView()
        XCTAssertNotNil(overlayView.navigationBar)
    }
    
    func testLeftBarButtonItemSetterAffectsNavigationBar() {
        let overlayView = NYTPhotosOverlayView()
        let leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: UIBarButtonSystemItem.Done, target: nil, action: nil)
        overlayView.leftBarButtonItem = leftBarButtonItem
        XCTAssert(leftBarButtonItem == overlayView.navigationBar.items?.first?.leftBarButtonItem)
    }
    
    func testRightBarButtonItemSetterAffectsNavigationBar() {
        let overlayView = NYTPhotosOverlayView()
        let rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: UIBarButtonSystemItem.Done, target: nil, action: nil)
        overlayView.rightBarButtonItem = rightBarButtonItem
        XCTAssert(rightBarButtonItem == overlayView.navigationBar.items?.first?.rightBarButtonItem)
    }
    
    func testTitleSetterAffectsNavigationBar() {
        let overlayView = NYTPhotosOverlayView()
        let title = "title"
        overlayView.title = title
        XCTAssert(title == overlayView.navigationBar.items?.first?.title)
    }
    
    func testTitleTextAttributesSetterAffectsNavigationBar() {
        let overlayView = NYTPhotosOverlayView()
        let titleTextAttributes = [NSForegroundColorAttributeName: UIColor.orangeColor()]
        overlayView.titleTextAttributes = titleTextAttributes
        XCTAssertEqual(titleTextAttributes, overlayView.navigationBar.titleTextAttributes as? [String: UIColor])
    }
}
